src/gun.erl

```diff
@@ -268,23 +305,35 @@ put(ServerPid, Path, Headers) ->
 put(ServerPid, Path, Headers, Body) ->
 	request(ServerPid, <<"PUT">>, Path, Headers, Body).
 
+-spec put(pid(), iodata(), headers(), iodata(), req_opts()) -> reference().
+put(ServerPid, Path, Headers, Body, ReqOpts) ->
+	request(ServerPid, <<"PUT">>, Path, Headers, Body, ReqOpts).
+
 -spec request(pid(), iodata(), iodata(), headers()) -> reference().
 request(ServerPid, Method, Path, Headers) ->
-	StreamRef = make_ref(),
-	_ = ServerPid ! {request, self(), StreamRef, Method, Path, Headers},
-	StreamRef.
+	request(ServerPid, Method, Path, Headers, <<>>, #{}).
 
 -spec request(pid(), iodata(), iodata(), headers(), iodata()) -> reference().
 request(ServerPid, Method, Path, Headers, Body) ->
+	request(ServerPid, Method, Path, Headers, Body, #{}).
+
+-spec request(pid(), iodata(), iodata(), headers(), iodata(), req_opts()) -> reference().
+request(ServerPid, Method, Path, Headers, Body, ReqOpts) ->
 	StreamRef = make_ref(),
-	_ = ServerPid ! {request, self(), StreamRef, Method, Path, Headers, Body},
+	ReplyTo = maps:get(reply_to, ReqOpts, self()),
+	_ = ServerPid ! {request, ReplyTo, StreamRef, Method, Path, Headers, Body},
 	StreamRef.
 
 %% Streaming data.
 
 -spec data(pid(), reference(), fin | nofin, iodata()) -> ok.
 data(ServerPid, StreamRef, IsFin, Data) ->
-	_ = ServerPid ! {data, self(), StreamRef, IsFin, Data},
+	data(ServerPid, StreamRef, IsFin, Data, #{}).
+
+-spec data(pid(), reference(), fin | nofin, iodata(), req_opts()) -> ok.
+data(ServerPid, StreamRef, IsFin, Data, ReqOpts) ->
+	ReplyTo = maps:get(reply_to, ReqOpts, self()),
+	_ = ServerPid ! {data, ReplyTo, StreamRef, IsFin, Data},
```

Comment:
Yeah, I don't see any real use case here. I will remove the `ReqOpts` argument for `data` and `cancel` functions.